### DISCUSSION TITLE
Authenticated HTTP calls between custom application services

### DISCUSSION THREAD

**Discussion by BlancoMY** [edgexfoundry/community#251] [Q&A] [answered]
Hello,

I am working on a project that involves creating custom application services using the EdgeX application-service-sdk. I want to make authenticated calls between these services using the JWT token-based authentication mechanism.

However, I noticed that the SDK does not expose the AddAuthenticationData or GetSelfJWT functions that are needed to obtain or add a JWT token to the HTTP requests. These functions are only available through the AuthenticationInjector or SecretProviderExt interfaces, which are not accessible from the SDK.

Is there a way to use these functions from the SDK, or is there another recommended way to make authenticated calls between custom application services using the application service identity? I would appreciate any guidance or examples on how to achieve this.

Thank you for your time and help.

**Comment by lenny-goodell** [comment, marked as answer]
Yes, the SecretProvider implementation is also a interfaces.SecretProviderExt, so you can cast the instance to be a SecretProviderExt.
```
sp = appSdk.SecretProvider()
spExt := sp.(interfaces.SecretProviderExt)
```

**Reply by BlancoMY** [reply]
That's working ! Thank you.

**Comment by cloudxxx8** [comment]
close as answered
